packages/matomo/index.tsx

```diff
@@ -1,13 +1,110 @@
+import { CartLineItem } from '@ifixit/cart-sdk';
+import { Money, sumMoney } from '@ifixit/helpers';
+
+const _paq: any[] | undefined =
+   typeof window === 'undefined' ? undefined : (window as any)._paq;
+
 /**
  * @see https://developer.matomo.org/api-reference/tracking-javascript
  */
+export function trackPageView(url: string) {
+   if (!_paq) {
+      return;
+   }
+   _paq.push(['setCustomUrl', url]);
+   _paq.push(['setDocumentTitle', document.title]);
+   _paq.push(['trackPageView']);
+}
 
-declare const _paq: string[][] | undefined;
+/**
+ * @see https://developer.matomo.org/api-reference/tracking-javascript
+ * @see https://matomo.org/docs/ecommerce-analytics/#tracking-product-views-in-matomo
+ */
+type ProductData = {
+   productSku: string;
+   productName?: string;
+   /**
+    * Category name, or up to five unique categories, e.g. ["Books", "New
+    * Releases", "Technology"]
+    */
+   categoryName?: string | [string, string?, string?, string?, string?];
+   price: Money;
+};
 
-export function trackPageView(url: string) {
-   if (typeof window !== 'undefined' && _paq) {
-      _paq.push(['setCustomUrl', url]);
-      _paq.push(['setDocumentTitle', document.title]);
-      _paq.push(['trackPageView']);
+export function trackMatomoEcommerceView(product: ProductData) {
+   if (!_paq) {
+      return;
+   }
+   _paq.push([
+      'setEcommerceView',
+      product.productSku,
+      product.productName,
+      product.categoryName,
+      product.price.amount,
+   ]);
+}
+
+export function trackMatomoCartChange(items: CartLineItem[]) {
+   if (!_paq) {
+      return;
+   }
+   trackClearCart();
+   if (items.length === 0) {
+      return;
+   }
+   items.forEach((item) => {
+      trackAddToCart({
+         productSku: item.itemcode,
+         productName: item.internalDisplayName,
+         price: item.price,
+         quantity: item.quantity,
+      });
+   });
+
+   const totalPrice = sumMoney(items.map((i) => i.price));
+   trackCartUpdated(totalPrice);
+}
+
+/**
+ * @see https://developer.matomo.org/api-reference/tracking-javascript
+ * @see https://matomo.org/docs/ecommerce-analytics/#example-of-adding-a-product-to-the-order
+ */
+type AddToCartData = Omit<ProductData, 'price'> & {
+   price?: Money;
```

Comment:
Why do we omit and re-add price?

Reply:
Price is a mandatory field for product view events (ie ProductData), and it's an optional field in add to cart events.

Reply:
![image](https://user-images.githubusercontent.com/52104630/195260188-02587c46-6b2d-4c19-b26e-f2541a666553.png)
https://developer.matomo.org/api-reference/tracking-javascript